mitiq/interface/mitiq_qiskit/transpiler.py

```diff
@@ -0,0 +1,114 @@
+# Copyright (C) Unitary Fund
+#
+# This source code is licensed under the GPL license (v3) found in the
+# LICENSE file in the root directory of this source tree.
+#
+# This module includes a modified version of the ApplyLayout class originally
+# licensed under the Apache License, Version 2.0.
+#
+# Original Author: IBM
+# Modification made by: André Alves on 07 Apr 2024
+#
+# For the original code and license, see
+# https://github.com/Qiskit/qiskit/blob/stable/1.0/qiskit/transpiler/passes/layout/apply_layout.py
```

Comment:
License-wise, it's all good. My concern about having this file in mitiq is that it's going to be hard to keep it synced with the upstream qiskit file. Can't we just use the imported functions as they are and wrap them when needed?

Reply:
The idea is not to keep them in sync. I used a lot of the original `ApplyLayout`, but I made substantive changes that are different from the original. The original class applies a `QuantumRegister` with the name `q`. We initialize the class with our own registers, and we use their names. From what I understand, maintaining the names is one of the purposes of the `_transform` function. But with probably half the class identical to the original, I thought it appropriate to include the license statement.